doc/nrf/security/tfm.rst

```diff
@@ -254,38 +269,469 @@ We will decrease the size of the (optional) ``mcuboot_pad`` partition and thus t
+Following sections describe the available TF-M partitions and their purpose.
+
+Platform partition
+------------------
+
+To enable platform partition, set the :kconfig:option:`CONFIG_TFM_PARTITION_PLATFORM` Kconfig option.
+It provides platform-specific services to other TF-M partitions and to a non-secure application.
 
+For the non-secure application, setting the :kconfig:option:`CONFIG_TFM_ALLOW_NON_SECURE_FAULT_HANDLING` Kconfig option enables more detailed error descriptions of faults from the non-secure application with the Zephyr fault handler.
 
-.. _ug_tfm_migrate:
+Platform partition also exposes the following |NCS| specific APIs for the non-secure application:
 
-Migrating from Secure Partition Manager to Trusted Firmware-M
-*************************************************************
+  .. code-block:: c
 
-The interface to TF-M is different from the interface to SPM.
-Due to that, the application code that uses the SPM Secure Services needs to be ported to use TF-M instead.
+    /* Search for the fw_info structure in firmware image located at address. */
+    int tfm_platform_firmware_info(uint32_t fw_address, struct fw_info *info);
 
-TF-M can replace the following SPM services:
+    /* Check if S0 is the active B1 slot. */
+    int tfm_platform_s0_active(uint32_t s0_address, uint32_t s1_address, bool *s0_active);
 
-* ``spm_request_system_reboot`` with ``tfm_platform_system_reset``.
-* ``spm_request_random_number`` with ``psa_generate_random`` or ``entropy_get_entropy``.
-* ``spm_request_read`` with ``tfm_platform_mem_read`` or ``soc_secure_mem_read``.
-* ``spm_s0_active`` with ``tfm_platform_s0_active``.
-* ``spm_firmware_info`` with ``tfm_firmware_info``.
+See :ref:`lib_tfm_ioctl_api` for more information about APIs available for the non-secure application.
 
-The following SPM services have no replacement in TF-M:
+For more information about the general features of the TF-M Platform partition, see `TF-M Platform`_.
 
-* ``spm_prevalidate_b1_upgrade``
-* ``spm_busy_wait``
-* ``spm_set_ns_fatal_error_handler``
+Internal Trusted Storage partition
+----------------------------------
+
+To enable the Internal Trusted Storage (ITS) partition, set the :kconfig:option:`CONFIG_TFM_PARTITION_INTERNAL_TRUSTED_STORAGE` Kconfig option.
+It implements the PSA Internal Trusted Storage APIs (`PSA Certified Secure Storage API 1.0`_) to achieve confidentiality, authenticity and encryption in rest (optional).
+ITS is meant to be used by the other TF-M partitions.
+It must not be accessed directly by the non-secure application.
+Instead, use the :ref:`tfm_partition_ps`.
+
+For more information about the general features of the TF-M ITS service, see `TF-M ITS`_.
+
+.. _tfm_encrypted_its:
+
+Encrypted ITS
+~~~~~~~~~~~~~
+
+To enable TF-M ITS encryption, set the :kconfig:option:`CONFIG_TFM_ITS_ENCRYPTED` Kconfig option.
+The ITS encryption is transparent to the user as long as the Master Key Encryption Key (MKEK) is populated before use.
```

Comment:
We need a brief definition of what the TF-M ITS encryption is before these lines. One cannot enable something that one doesn't know...
```suggestion
TF-M ITS encryption is a data protection mechanism in Internal Trusted Storage. It provides transparent encryption using a Master Key Encryption Key (MKEK) stored in hardware, with unique encryption keys derived for each file.

To enable TF-M ITS encryption, set the :kconfig:option:`CONFIG_TFM_ITS_ENCRYPTED` Kconfig option.
```